openedx/core/djangoapps/content/course_overviews/tests.py

```diff
@@ -256,3 +258,31 @@ def test_course_overview_caching(self, modulestore_type, min_mongo_calls, max_mo
         # we expect no modulestore queries to be made.
         with check_mongo_calls(0):
             _course_overview_2 = CourseOverview.get_from_id(course.id)
+
+    @ddt.data(ModuleStoreEnum.Type.split, ModuleStoreEnum.Type.mongo)
+    def test_get_non_existent_course(self, modulestore_type):
+        """
+        Tests that requesting a non-existent course from get_from_id raises
+        CourseOverview.DoesNotExist.
+        """
+        store = modulestore()._get_modulestore_by_type(modulestore_type)  # pylint: disable=protected-access
+        with self.assertRaises(CourseOverview.DoesNotExist):
+            CourseOverview.get_from_id(store.make_course_key('Non', 'Existent', 'Course'))
+
+    def test_get_errored_course(self):
+        """
+        Test that getting an ErrorDescriptor back from the module store causes
+        get_from_id to raise an IOError.
+        """
+        course = CourseFactory.create(
+            course="TEST101",
+            org="edX",
+            run="Run1",
+            mobile_available=True,
+        )
```

Comment:
Why was it necessary to pass in these arguments to `CourseFactory`? The default method call would suffice, no?